frontends/p4/fromv1.0/programStructure.cpp

```diff
@@ -143,6 +145,13 @@ void ProgramStructure::createTypes() {
         auto type = it.first->type;
         createType(type, true, &converted);
     }
+    for (auto it : registers) {
+        if (it.first->layout) {
+            auto type = types.get(it.first->layout);
+            if (type->is<IR::Type_Header>())
+                createType(type, true, &converted);
```

Comment:
It may be that the best place to do the conversion to a struct is somewhere earlier, in the P4-14 typechecker. I will try to figure it out.